provisioning/device/src/polling_state_machine.ts

```diff
@@ -139,18 +148,34 @@ export class  PollingStateMachine extends EventEmitter {
               this._fsm.transition('polling', request, operationId, pollingInterval, callback);
             }, pollingInterval);
           },
-          cancel: (err, result, response, callback) => {
+          cancel: (cancelledOpErr, callback) => {
             /* Codes_SRS_NODE_PROVISIONING_TRANSPORT_STATE_MACHINE_18_027: [ If a registration is in progress, `cancel` shall cause that registration to fail with an `OperationCancelledError`. ] */
             clearTimeout(this._pollingTimer);
             this._pollingTimer = null;
-            this._fsm.transition('cancelling', err, result, response, callback);
+            this._fsm.transition('cancelling', cancelledOpErr, callback);
+          },
+          /* Codes_SRS_NODE_PROVISIONING_TRANSPORT_STATE_MACHINE_18_034: [ If `disconnect` is called while the state machine is waiting to poll, the current operation shall be cancelled and the transport shall be disconnected. ] */
+          disconnect: (callback) => {
+            this._fsm.handle('cancel', new errors.OperationCancelledError(''), (err) => {
+              this._fsm.handle('disconnect', callback);
+            });
           },
           register: (request, callback) => callback(new errors.InvalidOperationError('another operation is in progress'))
         },
         polling: {
           _onEnter: (request, operationId, pollingInterval, callback) => {
+            /* Codes_SRS_NODE_PROVISIONING_TRANSPORT_STATE_MACHINE_18_037: [ If `PollingTransport.queryOperationStatus` does not call its callback within `ProvisioningDeviceConstants.defaultTimeoutInterval` ms, register shall with with a `TimeoutError` error. ] */
+            let timeoutTimer: any = setTimeout(() => {
+              debug('timeout while query');
+              if (this._currentOperationCallback === callback) {
+                debug('legit');
```

Comment:
>debug('legit') [](start = 16, length = 14)

those logs end up being what we use with customers in github issues. more detailed explanation?